### DISCUSSION TITLE
Hardcode custom attribute for all users coming from specific IdP

### DISCUSSION THREAD

**Discussion by cesdperez** [keycloak/keycloak#15275] [Q&A] [answered]
Hi!

I'm using Keycloak as an identity broker, I have set up multiple Identity Providers. 
I'd like to set a custom attribute for users that are logging in through a specific Identity Provider. Intuitively, I thought I could achieve this by creating an Identity Provider mapping of type "Hardcoded Attribute". It looks like this:
<img width="1051" alt="image" src="https://user-images.githubusercontent.com/16674679/199453457-e486fb53-80a7-40f3-b891-cffbf9454634.png">

It didn't work. I don't receive "my_custom_attribute" nor "abc123" anywhere inside `keycloak` after logging in (I'm using `keycloak-js`).

I couldn't find details on this in the documentation, sorry if I missed them.

Is using a hardcoded attribute the proper way to achieve this? If so, what I'm doing wrong?
Thanks a lot for your help.

Edit: when I say "anywhere inside `keycloak`", I also mean it's not coming in the idtoken nor accesstoken.

More details: This IdP configuration is using SAML, and I'm using Keycloak 18.0.2.

**Comment by vplemyannik** [comment, marked as answer]
@cesdperez did you add  OIDC UserAttribute mapper for your `keycloak-js` client ?
like this
![image](https://user-images.githubusercontent.com/17518385/199455852-125770ba-4921-40b1-bc25-abc30d4cb071.png)

**Reply by cesdperez** [reply]
I was missing that, indeed. Thanks for your help!

**Reply by vplemyannik** [reply]
You are welcome!
